<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1201</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>618</width>
      <height>432</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="vertsListWidget">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>50</y>
      <width>71</width>
      <height>171</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="halfEdgesListWidget">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>50</y>
      <width>71</width>
      <height>171</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="facesListWidget">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>50</y>
      <width>71</width>
      <height>171</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>220</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>220</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Half-Edges</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>220</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Faces</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>261</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>1081</x>
      <y>261</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>1141</x>
      <y>261</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>260</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex Pos</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>1019</x>
      <y>285</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>285</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>285</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>330</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Blue</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>1019</x>
      <y>330</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Red</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceBlueSpinBox">
    <property name="geometry">
     <rect>
      <x>1141</x>
      <y>310</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>310</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Face Color</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceGreenSpinBox">
    <property name="geometry">
     <rect>
      <x>1081</x>
      <y>310</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceRedSpinBox">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>310</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.050000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>330</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Green</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="chooseOBJ">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>10</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose OBJ</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addMidpoint">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>370</y>
      <width>91</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Midpoint</string>
    </property>
   </widget>
   <widget class="QPushButton" name="triangulateFace">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>410</y>
      <width>91</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Triangulate </string>
    </property>
   </widget>
   <widget class="QPushButton" name="catmullClark">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>370</y>
      <width>91</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Smooth</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadJSON">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>10</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose JSON</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>220</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Joints</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QTreeWidget" name="jointsTreeWidget">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>50</y>
      <width>281</width>
      <height>171</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string notr="true">1</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="meshSkinning">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>410</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Mesh Skinning</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>255</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Pos</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>285</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>260</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>286</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>260</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>260</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.989999999999995</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>285</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotXPlus">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>370</y>
      <width>41</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>x+5</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotYPlus">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>370</y>
      <width>41</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>y+5</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotZPlus">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>370</y>
      <width>41</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>z+5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>310</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Rotate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotXMinus">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>370</y>
      <width>41</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>x-5</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotYMinus">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>370</y>
      <width>41</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>y-5</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotZMinus">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>370</y>
      <width>41</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>z-5</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="jointRotX">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>312</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>341</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_19">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>340</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_20">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>340</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QTextBrowser" name="jointRotY">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>310</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="jointRotZ">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>310</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="exportUSD">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>10</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>export USD</string>
    </property>
   </widget>
   <widget class="QPushButton" name="changeColor">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>10</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Change Color</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1201</width>
     <height>36</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
